launcher/core/launcher.cpp

```diff
@@ -329,6 +330,13 @@ void Launcher::readyRead()
             msg >> d->serverAddress;
             break;
         }
+        case Protocol::ServerLaunchError:
+        {
+            QString reason;
+            msg >> reason;
+            QMessageBox::critical(nullptr, "GammaRay server could not start", "The server could not start: " + reason);
```

Comment:
This looks problematic in terms of dependencies, as the core part of the launcher code can also used without widgets even being available, e.g. think about the CLI launcher on an embedded target.